### DISCUSSION TITLE
Cannot create user with SDK

### DISCUSSION THREAD

**Discussion by lukepighetti** [pocketbase/pocketbase#5047] [Q&A] [answered]
Hi there,

I was unable to find a way to create a new user with the SDK

```dart
pb.collection('users').authWithPassword('test', 'test12345'); // throwing 400 error
```

I had to go into the admin panel and create a user with those exact credentials to be able to authenticate. My expectation based on the docs was that it would create the user automatically if it didn't yet exist.

Any insights? Did I miss a step?

<img width="715" alt="Screenshot 2024-06-07 at 5 58 13 PM" src="https://github.com/pocketbase/pocketbase/assets/666539/749ae657-b18f-42e3-8a11-7abcc0d0cab9">

<img width="706" alt="Screenshot 2024-06-07 at 5 58 24 PM" src="https://github.com/pocketbase/pocketbase/assets/666539/ec588889-c01f-4913-a506-a54bd9e1d7ca">

**Comment by ganigeorgiev** [comment]
Most likely some of your field constraints in failing. You can find more details about the failed request in the Admin UI > Logs and/or start the executable with `--dev`.

Assuming that you use the Dart SDK, you could also try to inspect the thrown exception - https://github.com/pocketbase/dart-sdk?tab=readme-ov-file#error-handling.

**Reply by ganigeorgiev** [reply, marked as answer]
Actually on second read you are not sending a create user request.

You need to call `pb.collection('users').create(...)`.

`authWithPassword` just authenticate, it doesn't create a user automatically.

The only exception to this is `authWithOAuth2` signup because otherwise we cannot link to the OAuth2 account.

**Reply by lukepighetti** [reply]
Thank you
